### ISSUE TITLE
Add support for ittp:aspectRatio

### ISSUE THREAD

**Issue by zcorpan** [w3c/webvtt#278]
See https://github.com/w3c/ttml-webvtt-mapping/issues/8

> The addition of ittp:aspectRatio [1] to IMSC1 was specifically requested to support display of the same caption over multiple aspect ratio versions of the same content.
>
> Can't this be supported in WebVTT?
>
> [1] https://dvcs.w3.org/hg/ttml/raw-file/tip/ttml-ww-profiles/ttml-ww-profiles.html#ttp-aspectRatio
>
> (raised by Pierre-Anthony Lemieux on 2015-09-23)
> From tracker issue http://www.w3.org/AudioVideo/TT/tracker/issues/440

**Comment by dwsinger** [comment]
I’m really unclear as to how this would be used.  Given that all our measurements are nominally related to the video region, isn’t the most we need a statement of the form “the dimensions here assumed a video region of X-wide by Y-high”?  Then one can notice that the actual video in hand is a different size and/or shape and adjust or panic as needed?


David Singer
Manager, Software Standards, Apple Inc.

**Comment by zcorpan** [comment]
IIUC, the desire is to author one subtitle file with positioning that maps to the content in the video, and use that file for all available variations of that video -- e.g. 4:3 video for 4:3 devices and 16:9 for 16:9 devices. The subtitle can then say that it only wants a 4:3 "window" for positioning, and then its positions will map to the same content in both videos.

But it's possible I'm off the mark here. cc @palemieux

**Comment by palemieux** [comment]
> IIUC, the desire is to author one subtitle file with positioning that maps to the content in the video, and use that file for all available variations of that video -- e.g. 4:3 video for 4:3 devices and 16:9 for 16:9 devices. The subtitle can then say that it only wants a 4:3 "window" for positioning, and then its positions will map to the same content in both videos.

Yes. The idea is to maintain the position of caption/subtitle relative to elements within each frame of the video program, e.g. to match the position of actors.

**Comment by silviapfeiffer** [comment]
Isn't that already solved by having percentage positioning?

**Comment by palemieux** [comment]
Assuming that the root container is mapped to the entire video image, 10% from the left edge of the root container mapped to a 4:3 image center-cut from a 16:9 image is not the same position (relative to the action on the image) as 10% from the left edge of the root container mapped to the aforementioned 16:9 image.

**Comment by dwsinger** [comment]
> On Dec 10, 2015, at 16:13 , Pierre-Anthony Lemieux <notifications@github.com> wrote:
> 
> Assuming that the root container is mapped to the entire video image, 10% from the left edge of the root container mapped to a 4:3 image center-cut from a 16:9 image is not the same position (relative to the action on the image) as 10% from the left edge of the root container mapped to the aforementioned 16:9 image.

True, but knowing the expected display ratio (“I designed this for the 16:9 version”) doesn’t help me, if I find this text track associated with the 4:3 version.  Was it letter-boxed or pan-scanned?  And so on.

I guess I am unclear as to how this attribute is expected to be used.

David Singer
Manager, Software Standards, Apple Inc.

**Comment by palemieux** [comment]
The use case is that of a content provider that wishes to use the same caption file for a given 16:9 video content, whether it is presented 16:9 or center-cut (4:3).

**Comment by silviapfeiffer** [comment]
Pardon my ignorance, but is centre cutting a standard way of conversion between aspect ratios in the TV and movie industry and online?

**Comment by nigelmegitt** [comment]
Yes, many TV display devices do this, and many content providers create their video with this in mind, concentrating the important parts of the image in a central 'safe area'.

**Comment by dwsinger** [comment]
> On Dec 10, 2015, at 21:04 , Pierre-Anthony Lemieux <notifications@github.com> wrote:
> 
> The use case is that of a content provider that wishes to use the same caption file for a given 16:9 video content, whether it is presented 16:9 or center-cut (4:3).

But…I still don’t get what algorithm the user-side is supposed to apply.  And I don’t get how it assumes center-cut, and not pan-scan or letterbox.

Imagine either a timed text file with dimensions in percentages, or with absolute dimensions but the overall dimensions known (so the percentages can be calculated). What does one do better/differently with that file if it also includes aspectRatio?


David Singer
Manager, Software Standards, Apple Inc.

**Comment by palemieux** [comment]
Please see the mapping algorithm at https://rawgit.com/w3c/imsc1/master/spec/ttml-ww-profiles.html#ttp-aspectRatio

**Comment by dwsinger** [comment]
"If ittp:aspectRatio is present, the root container shall be mapped to a rectangular area within the image frame such that:

the ratio of the width to the height of the rectangular area is equal to ittp:aspectRatio,
the center of the rectangular area is collocated with the center of the image frame,
the rectangular area is entirely within the image frame, and
the rectangular area has a height or width equal to that of the image frame."

OK, this deals with center-cropped; imagine I find a 4:3 video with captions designed for 16:9. How do I know if it was center-cropped, pan-scanned, or letterboxed? Certainly for pan-scan, the actiev area has moved around the 16:9 video and so any static decision is necessarily wrong.

Still puzzled.  This seems to ask me to make a 16:9 rectangle within the 4:3 I see and position the captions within that, which doesn't sound right to me. What am I missing?
